packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -61,18 +59,20 @@ export class IpfsCredentialResolver implements CredentialResolver {
       | VerifiableCredential<RoleCredentialSubject>
       | RoleEIP191JWT
       | undefined;
-    const cachedRoleCredential = this._dataAggregator.getRoleCredential(
+    const cachedRoleCredential = this._roleCredentialCache?.getRoleCredential(
       did,
       namespace
     );
     if (cachedRoleCredential) {
-      return cachedRoleCredential.data;
+      return cachedRoleCredential;
     }
     credential = await this.getVerifiableCredential(did, namespace);
     if (!credential) {
       credential = await this.getEIP191JWT(did, namespace);
     }
```

Comment:
This is a nice idea, but if you consider an issuer-verification call then if the credential is not there it would return with a unverified status or throw error. So caching a non-existent credential doesn't really add a value IMO.

Reply:
We can throw only when there is `null`. When credential is `undefined` then we need to fetch from chain

Reply:
That behaviour is already there, if it is undefined then credential is being fetched from chain.